util/fipstools/inject_hash/macho_parser/macho_parser.c

```diff
@@ -0,0 +1,165 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0 OR ISC
+
+#include <stdint.h>
+
+#include "common.h"
+#include "macho_parser.h"
+
+// Documentation for the Mach-O structs can be found in macho-o/loader.h and mach-o/nlist.h
+int read_macho_file(const char *filename, machofile *macho) {
+    FILE *file = NULL;
+    load_cmd *load_commands = NULL;
+    int ret = 1;
+
+    file = fopen(filename, "rb");
+    if (!file) {
+        LOG_ERROR("Error opening file %s", filename);
+        ret = 0;
+        goto end;
+    }
+
+    fread(&macho->macho_header, sizeof(macho_header), 1, file);
+    if(macho->macho_header.magic != MH_MAGIC_64) {
+        LOG_ERROR("File is not a 64-bit Mach-O file");
+        ret = 0;
+        goto end;
+    }
+
+    load_commands = malloc(macho->macho_header.sizeofcmds);
+    fread(load_commands, macho->macho_header.sizeofcmds, 1, file);
+
+    // We're only looking for __text, __const in the __TEXT segment, and the string & symbol tables
+    macho->num_sections = 4;
+    macho->sections = malloc(macho->num_sections * sizeof(section_info));
+
+    uint32_t section_index = 0;
+    for (uint32_t i = 0; i < macho->macho_header.sizeofcmds / BIT_MODIFIER; i += load_commands[i].cmdsize / BIT_MODIFIER) {
+        if (load_commands[i].cmd == LC_SEG) {
```

Comment:
mmmh, your tests just use `LC_SEGMENT_64`?

Reply:
There's a macro defining it:
```
#define LC_SEG LC_SEGMENT_64
```